app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -254,9 +258,67 @@ class SubmissionFileManager extends React.Component {
     );
   };
 
+  renderRequiredFiles = () => {
+    let requiredFilesBox;
+
+    if (this.state.requiredFiles.length > 0) {
+      requiredFilesBox = (
+        <div>
+          <h2>{I18n.t("activerecord.attributes.assignment.assignment_files")}</h2>
+          <p>
+            {this.state.numberOfMissingFiles === 0 ? (
+              <p>{I18n.t("student.submission.all_files_submitted")}</p>
+            ) : (
+              I18n.t("student.submission.missing_files", {file: this.state.numberOfMissingFiles})
+            )}
+          </p>
+          {this.state.requiredFiles.map(filename => {
+            return (
+              <div key={filename} className={"required-files-container"}>
+                <input
+                  className={"required-files-checkbox"}
+                  value={filename}
```

Comment:
You don't need to give this checkbox a `value` attribute. However, you _should_ give this checkbox a `name` attribute (maybe of the form `required-file-${filename}`) so that it can be associated with a label (see comment below).